services/blockchain-connector/shared/sdk/encoder.js

```diff
@@ -17,54 +17,51 @@ const { codec } = require('@liskhq/lisk-codec');
 
 const { parseInputBySchema } = require('../parser');
 const {
-	getAccountSchema,
 	getBlockSchema,
 	getBlockHeaderSchema,
-	getBlockHeaderAssetSchema,
+	getBlockAssetSchema,
 	getTransactionSchema,
-	getTransactionAssetSchema,
+	getTransactionParamsSchema,
 } = require('./schema');
 
-const encodeAccount = async (account) => {
-	const accountSchema = await getAccountSchema();
-	const accountBuffer = codec.encode(accountSchema, account);
-	return accountBuffer.toString('hex');
-};
-
 const encodeTransaction = async (transaction) => {
-	const txAssetSchema = await getTransactionAssetSchema(transaction);
-	const txAssetBuffer = codec.encode(txAssetSchema, transaction.asset);
+	const txParamsSchema = await getTransactionParamsSchema(transaction);
+	const txParamsBuffer = codec.encode(txParamsSchema, transaction.asset);
 
 	const txSchema = await getTransactionSchema();
 	const txBuffer = codec.encode(
 		txSchema,
-		{ ...transaction, asset: txAssetBuffer },
+		{ ...transaction, params: txParamsBuffer },
 	);
 
 	return txBuffer.toString('hex');
 };
 
 const encodeBlock = async (block) => {
-	const { header, payload } = block;
+	const { header, assets, transactions } = block;
 
-	// Handle the block payload
-	const blockPayloadBuffer = await Promise.all(payload.map(tx => encodeTransaction(tx)));
+	// Handle the block transactions
+	const blockTransactionsBuffer = await Promise.all(transactions.map(tx => encodeTransaction(tx)));
 
-	// Handle the block header asset
-	const blockHeaderAssetSchema = await getBlockHeaderAssetSchema(header.version);
-	const parsedHeaderAsset = parseInputBySchema(header.asset, blockHeaderAssetSchema);
-	const blockHeaderAssetBuffer = codec.encode(blockHeaderAssetSchema, parsedHeaderAsset);
+	// Handle the block assets
+	const blockAssetSchema = await getBlockAssetSchema();
+	const parsedAssets = await Promise
+		.all(assets.map(asset => parseInputBySchema(asset, blockAssetSchema)));
+	const blockAssetBuffer = await Promise
+		.all(parsedAssets.map(asset => codec.encode(blockAssetSchema, asset)));
 
 	// Handle the block header
 	const blockHeaderSchema = await getBlockHeaderSchema();
 	const blockHeaderBuffer = codec.encode(
 		blockHeaderSchema,
 		{
 			...header,
-			asset: blockHeaderAssetBuffer,
 			previousBlockID: Buffer.from(header.previousBlockID, 'hex'),
 			transactionRoot: Buffer.from(header.transactionRoot, 'hex'),
-			generatorPublicKey: Buffer.from(header.generatorPublicKey, 'hex'),
+			generatorAddress: Buffer.from(header.generatorAddress, 'hex'),
+			assetsRoot: Buffer.from(header.assetsRoot, 'hex'),
+			stateRoot: Buffer.from(header.stateRoot, 'hex'),
+			validatorsHash: Buffer.from(header.validatorsHash, 'hex'),
 			signature: Buffer.from(header.signature, 'hex'),
```

Comment:
Please check if this piece of code could be handled dynamically using the `parseInputBySchema` method.